### DISCUSSION TITLE
Does Prisma have Specification pattern like JPA?

### DISCUSSION THREAD

**Discussion by ngoc199** [prisma/prisma#9350] [Q&A] [answered]
Hi,

I want to make a search with custom fields like `title`, `author`, `created date`. I want to use the Specification Pattern, so I don't have to define multiple functions but `findAll(someSpecifications)`

How could I do this using Prisma?

**Comment by ryands17** [comment, marked as answer]
@ngoc199 👋 

Unfortunately this isn't possible as Prisma isn't a class based ORM so you cannot use the Specification Pattern like JPA. I would suggest watching this request for more updates 🙂 
https://github.com/prisma/prisma/issues/5273

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
